samples/Gaussian_initial_state/gauss_wavefcn.qs

```diff
@@ -0,0 +1,212 @@
+    function Norm (sigma: Double, mu: Double, N: Int) : Double {
+        mutable sum = 0.;
+        for (n in -N..N) {
+            set sum = PlusD(sum, NormTerm(sigma, mu, n));
+        }
+        return sum;
+    }
+
+    /// # Summary
+    /// Computes the rotation angle.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation.
+    /// ## mu
+    /// Mean.
+    /// ## N
+    /// The limit of the sum in the normalization factor.
+    operation Angle (sigma: Double, mu: Double, N : Int) : Double {
+        return ArcCos(Sqrt(Norm(sigma/2., mu/2., N)/Norm(sigma, mu, N)));
+    }
+
+    /// # Summary
+    /// Return a list of n-bit strings.
+    /// # Input
+    /// ## nQubits
+    /// The number of bits.
+    operation QubitStrings (nQubits: Int) : Bool[][] {
+        return MappedOverRange(0..PowI(2, n) - 1, IntAsBoolArray(_, n));
+    }
+
+    
+    /// # Summary
+    /// Given an n-bit string, return the corresponding mean for the rotation angle
+    /// at recursion level n.
+    /// # Input
+    /// ## qub
+    /// The n-bit string.
+    /// ## mu
+    /// Mean.
+    operation MeanQubitCombo (qub: Bool[], mu: Double) : Double {
+        mutable mu_out = mu;
+        for (bit in qub) {
+            set mu_out += mu_out/2. - bit ? 0.5 | -0.5;
+        }
+        return mu_out;
+    }
+
+    /// # Summary
+    /// At recursion level n, return a list of all the means used for the various rotation angles.
+    /// # Input
+    /// ## mu
+    /// Mean.
+    /// ## n
+    /// Recursion level.
+    operation LevelMeans (mu: Double, n: Int) : Double[] {
+        mutable list_mu_out = ConstantArray(2^n, 0.);
+        let qb_strings = QubitStrings(n);
+        for (i in 0..2^n-1) {
+            let mu_out = MeanQubitCombo(ElementAt(i, qb_strings), mu);
+            set list_mu_out w/= i <- mu_out;
+        }
+        return list_mu_out;
+    }
+
+    
+    /// # Summary
+    /// At recursion level n, return a list of all the rotation angles.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation
+    /// ## mu
+    /// Mean.
+    /// ## n
+    /// Recursion level.
+    operation LevelAngles(sigma: Double, mu: Double, n: Int) : Double[] {
+        let sigma_out = sigma/(2.^IntAsDouble(n));
+        let list_mu = LevelMeans(mu, n);
+        mutable angles_out = ConstantArray(2^n, 0.);
+        for (i in IndexRange(list_mu)) {
+            mutable mu_ = ElementAt(i, list_mu);
+            set angles_out w/= i <- Angle(sigma_out, mu_, 10^3);
+        }
+        return angles_out;
+    }
+
+    /// # Summary
+    /// Prepare the Gaussian wavefunction on a register.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation.
+    /// ## mu_
+    /// Mean.
+    /// ## num_qubits
+    /// The number of qubits.
+    operation PrepareGaussWavefcn (sigma: Double, mu_: Double, numQubits: Int, register: Qubit[]) : Unit is Adj {
+        // Compute angle.
+        mutable theta = Angle(sigma, mu_, 10^3);
```

Comment:
It shouldn't be possible to use `mutable` variables in an operation that is declared `Adj` without an explicit `adjoint` body.